liferay-content-management/liferay-portal
# LPD-26453 Success message is shown when saving AL on membership when user has not ASSIGN_MEMBERS permissions

LPD-26453

# What is this trying to solve?

https://liferay.atlassian.net/browse/LPD-26453


Here there was 2 problems. 
the organization portlet was not included on the command of the asset library
when adding an asset library to the user via the _Reproduction Steps_ the user does not have assign member permissions so it can not save it but the display context deleted the data so the user did not knew that something was not correct.

# How am I fixing it?
first adding the proper portlet to the action command
validating the permissions of the groups on the action and sending an exception if there is no proper permission.


# How can you verify that it works?
Reproduction Steps:

1) Open the main menu on the upper left, go to Applications --> Content --> Asset Libraries, and add an Asset Library.
2) At the Sites tab, connect the Asset Library to the DXP Site.
3) Go to Control Panel --> Users --> Users and Organizations --> Organizations, and create an Organization.
4) Create a User and add them to the created Organization.
5) Under Roles, add the Organization Owner Role to the User.
6) Under Memberships, add the DXP Site membership to the User.
7) Go to the DXP Site and make sure that the Site is configured as "Restricted" and "Allow Manual Membership Management" is enabled, by going to Configuration --> Site Settings --> Site Configuration.
8) Go to Users and Organizations and under the Users tab, impersonate the new User by clicking on the kebab menu beside them, then selecting "Impersonate User"
9) Under the User Profile Icon on the top right, select “My Organizations”.
10) Click on the Organization and then click on the User.
11) Go to the "Asset Libraries" tab, and click on "Select".
12) click on save
